travis_ci/configure-for-travis-vm

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+# This is a configure line for travis, which assumes that the
+# following Ubuntu trusty packages have been installed:
+# g++-4.9
+# gfortran-4.9
+# libblas-dev
+# liblapack-dev
+# libopenmpi-dev
+# libhdf5-serial-dev
+# environment-modules
+#
+# and also that googletest has been installed via spack.
+
+gtestdir=$( spack find -p googletest | tail -1 | awk '{print $2}' )
+
+./configure \
+	    --with-CXX=/usr/bin/mpic++ \
+	    --with-hdf5=/usr/bin/h5cc \
+	    --with-lapack='-lopenblas' \
+	    --with-gtest=${gtestdir}
```

Comment:
In light of some of the memory bugs I'm seeing, it may be wise to also add `--enable-opt=yes \ CXXFLAGS="-fsanitize=address"` (or similar). [Address Sanitizer](https://github.com/google/sanitizers/wiki/AddressSanitizer)(ASan) can be used to detect memory leaks, even on optimized code. I wouldn't recommend running ASan on code with compiler optimization flags in a debugger when trying to find and fix a memory leak (unless, of course, the bug does not occur when running with `-O0`), but in this case, we don't care about stepping through the code -- we only care whether or not memory leaks are present when running the tests on a CI server. ASan is available in GCC as of version 4.8 and in LLVM-based compilers as of LLVM version 3.1, so it should be possible to run our tests with ASan on Travis CI. ASan should slow execution by ~2x and can be run on optimized code, as opposed to Valgrind/memcheck, which slows execution by ~20x and its authors recommend not using it on optimized code. Hopefully, this performance penalty doesn't slow down execution to the point where testing exceeds Travis CI limits; otherwise, we'll have to figure out workarounds.